hooks/playbooks/adoption_deploy_ceph.yml

```diff
@@ -0,0 +1,143 @@
+---
+# Copyright Red Hat, Inc.
+# All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License. You may obtain
+# a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+
+- name: Deploy ceph for 17.1 deployment
+  hosts: "{{ cifmw_target_host | default('localhost') }}"
+  tasks:
+    - name: Gather ansible_user_dir from undercloud
+      delegate_to: "osp-undercloud-0"
+      ansible.builtin.setup:
+        gather_subset:
+          - user_dir
+
+    - name: Deploy 17.1 ceph
+      vars:
+        _cifmw_basedir_undercloud: >-
+          {{
+            [ansible_user_dir,
+            'ci-framework-data'
+            ] | path_join
+          }}
+        stack: >-
+          {{
+            cifmw_adoption_osp_deploy_scenario.stacks[stack_index | int]
+          }}
+        _overcloud_name: >-
+          {{
+            stack.stackname
+          }}
+        _roles_file: >-
+          {{
+            [cifmw_adoption_source_scenario_path,
+             stack.roles_file
+            ] | path_join
+          }}
+        _roles_file_dest: "{{ ansible_user_dir }}/roles_{{ _overcloud_name }}.yaml"
+        _network_data_file_dest: "{{ ansible_user_dir }}/network_data_{{ _overcloud_name }}.yaml"
+        _ceph_osd_spec_file: >-
+          {{
+            [cifmw_adoption_source_scenario_path,
+             stack.ceph_osd_spec_file
+            ] | path_join
+          }}
+        _ceph_osd_spec_file_name: "{{ _ceph_osd_spec_file | basename }}"
+        _ceph_osd_spec_file_dest: "{{ ansible_user_dir }}/{{ _ceph_osd_spec_file_name }}"
+        _cloud_domain: "{{ cifmw_adoption_osp_deploy_scenario.cloud_domain }}"
+        _source_cmd: "source {{ ansible_user_dir }}/stackrc"
+      block:
+        - name: Copy ceph osd file
+          delegate_to: "osp-undercloud-0"
+          ansible.builtin.copy:
```

Comment:
File manipulations without explicitly setting the `mode` will cause linter issues. iirc it's one of the rules we disabled, so no problem for now, but at some point someone will need to touch this freshly added code to make it complaint.

Reply:
good catch, I haven't thought about this, I think I added the `mode` to all the tasks now